### PR TITLE
Olga _ updated error messages on login page

### PR DESCRIPTION
**Pull request by cvtqx**
# Description
The error message on the login screen should be improved for clarity. 
When the username is not present in the system, it should display ""Username not found"" as the error message. 
Similarly, if the error is related to an incorrect password, the appropriate error message should be displayed.
Currently for both the scenarios the same error message is thrown “Invalid email and/ or password.”"

## Main changes explained:
Updated Logincontroller.js to include different error messages for non-existent user and for incorrect password.

## How to test:
1. check into current branch on the back end
2. do `npm run build` and `npm start` to run this PR locally
3. check into development branch on the front end
4. do `npm run start:local` to run the frontend
5. try to log in using an incorrect email (and any password). After clicking Submit button, you should see a message 'Username not found'
6. try to log in using a correct email and an incorrect password. After clicking Submit button, You should see a message 'Invalid password'.

## Screenshots or videos of changes:

Error message before:

Before_errorMessage

Error messages for invalid email and incorrect password after:

Non existent email:
After_emailField

Incorrect password:
After_passwordField

### CODE REVIEW

**Review by shiva2096 — APPROVED** [review]
Hi @cvtqx 

Great job! Functionality is working as expected.
1) Tested with un-registered user
2) Tested with registered user and wrong password
3) Tested with registered user and correct password


https://github.com/OneCommunityGlobal/HGNRest/assets/26687177/797933cc-6da8-4dbf-b300-9736d1867807

**Comment by mounica-dingari-by** [comment]
Hi @cvtqx , I can see that, if the username/password are incorrect or even if the user is inactive, the same error message "Invalid email and/or password" is thrown. I'm new to this, so please correct me if I'm wrong about this functionality. 


Screenshot from 2023-07-15 13-01-44

**Review by mounica-dingari-by — APPROVED** [review]

**Review by n0vay — APPROVED** [review]
Tested all Changes, everything works perfectly. Great Job.





image

**Review by vemanamadhu — APPROVED** [review]
The changes look good. While testing I can see the different error messages for different scenarios. So it is working as desired.

https://github.com/OneCommunityGlobal/HGNRest/assets/13665284/3be4e97c-b042-465c-b646-c13ebeeacc8f

**Review by siddharthmohite — APPROVED** [review]
This works as Intended.
Amazing!
Screenshot 2023-07-15 at 6 27 06 PM


Screenshot 2023-07-15 at 6 26 40 PM

**Review by AaronPersaud — APPROVED** [review]
Hi, I have tested the change and it looks great!

<b>Testing new login messages</b>

https://github.com/OneCommunityGlobal/HGNRest/assets/10265513/dc7d30d9-960f-45f4-b40e-c30a75ba839c

- logging in with invalid email 
- logging in with valid email and wrong password
- logging in with valid email and correct password

Nice work!

**Comment by AaronPersaud** [comment]
@one-community this PR has 5 approvals

File: src/controllers/logincontroller.js [hunk 1]
```diff
@@ -21,8 +21,13 @@ const logincontroller = function () {
...
 
-    // returning 403 if the user not found or the found user is inactive.
-    if (!user || user.isActive === false) {
+    // returning 403 if the user not found
+    if (!user) {
+      res.status(403).send({ message: 'Username not found.' });
+      return;
+    }
+    // returning 403 if the found user is inactive (Message is taken care of in PR438).
+    if (user.isActive === false) {
```
Comment (joyceiphone) [inline src/controllers/logincontroller.js:30]:
```suggestion
    if (!user.isActive) {
```

**Review by joyceiphone — APPROVED** [review]
Good work! Tested with invalid password and correct email address/ invalid email address and password. Some minor suggestion added.

Screen Shot 2023-07-17 at 7 36 29 AM
Screen Shot 2023-07-17 at 7 37 02 AM

File: src/controllers/logincontroller.js (hunk 1 above)
Comment (cvtqx) [reply src/controllers/logincontroller.js:30]: Thanks for the suggestion @joyceiphone ! I believe PR#438 is taking care of isActive status and a relevant message so I did not touch this line of code.

**Comment by cvtqx** [comment]
> Hi @cvtqx , I can see that, if the username/password are incorrect or even if the user is inactive, the same error message "Invalid email and/or password" is thrown. I'm new to this, so please correct me if I'm wrong about this functionality.
> 
> Screenshot from 2023-07-15 13-01-44

@mounica-dingari-by I see that you've approved it, so have you figured it out? If not, feel free to slack me and I'll try to help!

File: src/controllers/logincontroller.js (hunk 1 above)
Comment (joyceiphone) [reply src/controllers/logincontroller.js:30]: @cvtqx Okay. Thank you for letting me know.

**Comment by one-community** [comment]
Thank you everyone. Resolve the merge conflicts and this and I'll move it to final review.

**Review by BaileyM96 — APPROVED** [review]
Good afternoon @cvtqx I tested your changes and everything works great!

## Video Walkthrough 

[PR #441 Updated Error Message on Login.webm](https://github.com/OneCommunityGlobal/HGNRest/assets/108101478/dba3cca4-b8d0-474d-85bf-2d69bcf8f071)

**Review by luisarevalo21 — APPROVED** [review]
Hello @cvtqx ,

I tested your PR and works great.
1. if the user isn't in the system I get "username not found"
2. If the user type their password incorrectly I get "invalid password" 
3.  I also tested for an admin account and a volunteer account
Great job!

Screenshot 2023-07-18 at 8 37 11 PM
Screenshot 2023-07-18 at 8 37 25 PM
Screenshot 2023-07-18 at 8 37 37 PM

**Review by EdwinLauM — APPROVED** [review]
Hi @cvtqx! Works as intended, good job! 

https://github.com/OneCommunityGlobal/HGNRest/assets/42288987/f2ed305b-86d9-4d36-b7bc-feedf212741a

**Pull request merged**